client/state/posts/test/reducer.js

```diff
@@ -366,4 +370,105 @@ describe( 'reducer', () => {
 			expect( state ).to.eql( {} );
 		} );
 	} );
+
+	describe( '#siteRequests()', () => {
+		it( 'should default to an empty object', () => {
+			const state = siteRequests( undefined, {} );
+
+			expect( state ).to.eql( {} );
+		} );
+
+		it( 'should map site ID, post ID to true value if request in progress', () => {
+			const state = siteRequests( undefined, {
+				type: POST_REQUEST,
+				siteId: 2916284,
+				postId: 841
+			} );
+
+			expect( state ).to.eql( {
+				2916284: {
+					841: true
+				}
+			} );
+		} );
+
+		it( 'should accumulate mappings', () => {
+			const state = siteRequests( Object.freeze( {
+				2916284: {
+					841: true
+				}
+			} ), {
+				type: POST_REQUEST,
+				siteId: 2916284,
+				postId: 413
+			} );
+
+			expect( state ).to.eql( {
+				2916284: {
+					841: true,
+					413: true
+				}
+			} );
+		} );
+
+		it( 'should map site ID, post ID to false value if request finishes successfully', () => {
+			const state = siteRequests( Object.freeze( {
+				2916284: {
+					841: true
+				}
+			} ), {
+				type: POST_REQUEST_SUCCESS,
+				siteId: 2916284,
+				postId: 841
+			} );
+
+			expect( state ).to.eql( {
+				2916284: {
+					841: false
+				}
+			} );
+		} );
+
+		it( 'should map site ID, post ID to false value if request finishes with failure', () => {
```

Comment:
might want to keep around the reason it failed, at least for debugging? false doesn't tell us much.